frontend/src/app/pages/exercises/exercise/shared/trainer-map-editor/trainer-map-editor.component.ts

```diff
@@ -82,4 +90,28 @@ export class TrainerMapEditorComponent {
             template: vehicleTemplate,
         });
     }
+
+    public importingTemplates = false;
+    public async importPartialExport(fileList: FileList) {
+        try {
+            this.importingTemplates = true;
+            const importedText = await fileList.item(0)?.text();
+            if (importedText === undefined) {
+                // The file dialog has been aborted.
+                return;
+            }
+            const importedPlainObject = JSON.parse(importedText) as object;
+            const importedInstance = plainToInstance(
+                PartialExport,
+                importedPlainObject
+            );
+            const migratedInstance = migratePartialExport(importedInstance);
```

Comment:
I just noticed that this could not fail. However, I don't know whether this is good behavior. When importing an invalid file, the migration function will choose `undefined` for all three categories, which is valid, just not useful. I'm not quite sure how to proceed there.